src/sentry/receivers/outbox/control.py

```diff
@@ -37,13 +41,33 @@ def process_integration_updates(object_identifier: int, region_name: str, **kwds
 
 @receiver(process_control_outbox, sender=OutboxCategory.SENTRY_APP_UPDATE)
 def process_sentry_app_updates(object_identifier: int, region_name: str, **kwds: Any):
+
     if (
         sentry_app := maybe_process_tombstone(
             model=SentryApp, object_identifier=object_identifier, region_name=region_name
         )
     ) is None:
         return
-    sentry_app  # Currently we do not sync any other sentry_app changes, but if we did, you can use this variable.
+
+    # When a sentry app's definition changes purge cache for all the installations.
+    # This could get slow for large applications, but generally big applications don't change often.
+    install_query = SentryAppInstallation.objects.filter(sentry_app=sentry_app).values(
+        "id", "organization_id"
+    )
+    install_map: dict[int, int] = {}
+    for row in install_query:
+        install_map[row["organization_id"]] = row["id"]
+
```

Comment:
I don't see an index in the model preventing us from having multiple sentry app installations for a single organization/sentry app pair 😬 Do we need to handle this possible case?

Reply:
Good catch, I thought there was a constraint, but there isn't. And even worse, there *are* duplicates in prod.